src/llmq/quorums_instantsend.cpp

```diff
@@ -904,10 +880,33 @@ void CInstantSendManager::ProcessInstantSendLock(NodeId from, const uint256& has
         creatingInstantSendLocks.erase(islock->GetRequestId());
         txToCreatingInstantSendLocks.erase(islock->txid);
 
-        CInstantSendLockPtr otherIsLock;
-        if (db.GetInstantSendLockByHash(hash)) {
+        if (pendingInstantSendLocks.count(hash) || db.KnownInstantSendLock(hash)) {
             return;
         }
+    }
+
+    CTransactionRef tx;
+    uint256 hashBlock;
+    const CBlockIndex* pindexMined{nullptr};
+    // we ignore failure here as we must be able to propagate the lock even if we don't have the TX locally
+    if (GetTransaction(islock->txid, tx, Params().GetConsensus(), hashBlock) && !hashBlock.IsNull()) {
+        {
+            LOCK(cs_main);
+            pindexMined = LookupBlockIndex(hashBlock);
+        }
+
+        // Let's see if the TX that was locked by this islock is already mined in a ChainLocked block. If yes,
+        // we can simply ignore the islock, as the ChainLock implies locking of all TXs in that chain
+        if (llmq::chainLocksHandler->HasChainLock(pindexMined->nHeight, pindexMined->GetBlockHash())) {
```

Comment:
We should probably assert that pindexMined isn't null, though that's arguable outside of this PRs scope